### PR TITLE
Simple Payments: Use empty-content in upgrade jetpack dialog

### PR DESCRIPTION
**Pull request by donpark**
This PR enhances Upgrade Jetpack dialog using `empty-content` as mentioned in PR #16825 and `banner`.

screenshot_333

Test Plan:

1. Start editing a post on site with Jetpack version older than 5.2
2. Try using "Add Payment Button" menu item.
3. You should see the dialog above.
4. Clicking on the upgrade banner should navigate to Jetpack plugin page.

### CODE REVIEW

File: client/components/tinymce/plugins/simple-payments/dialog/index.jsx [hunk 1]
```diff
@@ -381,11 +384,25 @@ class SimplePaymentsDialog extends Component {
...
 		if ( ! shouldQuerySitePlans && isJetpackNotSupported ) {
 			return this.renderEmptyDialog(
-				<Notice
-					status="is-error"
-					text={ translate( 'Please upgrade to Jetpack 5.2 to use Simple Payments feature' ) }
-					onDismissClick={ onClose }
-				/>
+				<EmptyContent
+					illustration="/calypso/images/illustrations/illustration-jetpack.svg"
+					title={ translate( 'Upgrade Jetpack to use Simple Payments feature' ) }
```
Comment (a8ci18n) [inline client/components/tinymce/plugins/simple-payments/dialog/index.jsx]: :disappointed: 18 existing translations will be lost with this change.

**Comment by iamtakashi** [comment]
Noticed the text inside the banner has underline. Let's remove it.

Can the space between the image and the title be smaller?

**Comment by donpark** [comment]
@iamtakashi removed link underline and reduced spacing between illustration and title.

Illustrations' whitespace padding seems to vary a lot.

File: client/components/tinymce/plugins/simple-payments/dialog/index.jsx [hunk 2]
```diff
@@ -381,11 +384,25 @@ class SimplePaymentsDialog extends Component {
...
 			return this.renderEmptyDialog(
-				<Notice
-					status="is-error"
-					text={ translate( 'Please upgrade to Jetpack 5.2 to use Simple Payments feature' ) }
-					onDismissClick={ onClose }
-				/>
+				<EmptyContent
+					className="upgrade-jetpack"
+					illustration="/calypso/images/illustrations/illustration-jetpack.svg"
+					title={ translate( 'Upgrade Jetpack to use Simple Payments feature' ) }
```
Comment (a8ci18n) [inline client/components/tinymce/plugins/simple-payments/dialog/index.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Please upgrade to Jetpack 5.2 to use Simple Payments feature' )` *ES Score: 12*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by iamtakashi** [comment]
Looks good! Let's wait for the Editorial team for the copy review.

**Comment by iamtakashi** [comment]
Pinging @Automattic/editorial for the copy review since we're very close to the announcement.

**Comment by michelleweber** [comment]
Copy -- two small things:
- in the headline, let's say "the Simple Payments feature" or just "Simple Payments." (My choice would be just "Simple Payments.")
- Remove "feature" in the subhead as well, so it's just "Simple Payments requires..."

Since Simple Payments is in capital letters, it signals that it's a proper name, so it's neater and cleaner to nix "feature."

**Comment by iamtakashi** [comment]
Thanks you, @michelleweber !!

File: client/components/tinymce/plugins/simple-payments/dialog/index.jsx [hunk 3]
```diff
@@ -381,11 +384,23 @@ class SimplePaymentsDialog extends Component {
...
 			return this.renderEmptyDialog(
-				<Notice
-					status="is-error"
-					text={ translate( 'Please upgrade to Jetpack 5.2 to use Simple Payments feature' ) }
-					onDismissClick={ onClose }
-				/>
+				<EmptyContent
+					className="upgrade-jetpack"
+					illustration="/calypso/images/illustrations/illustration-jetpack.svg"
+					title={ translate( 'Upgrade Jetpack to use Simple Payments' ) }
```
Comment (a8ci18n) [inline client/components/tinymce/plugins/simple-payments/dialog/index.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Please upgrade to Jetpack 5.2 to use Simple Payments feature' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**